src/main/kotlin/com/manimdsl/linearrepresentation/LinearRepresentation.kt

```diff
@@ -75,9 +76,10 @@ data class StackPopObject(
     }
 }
 
-data class ArrayElemAssignObject(val arrayIdent: String, val index: Int, val newElemValue: ExecValue) : ManimInstr {
+data class ArrayElemAssignObject(val arrayIdent: String, val index: Int, val newElemValue: ExecValue, val animatedStyle: AnimationProperties?) : ManimInstr {
```

Comment:
I realise this is quite tightly coupled and not using the `RestyleObject`, but that is owing to the current nature of our `replace_text` function (see line 82). Am welcome to any suggestions on how to get this working more cleanly

Reply:
Also see `rectangle.py`

Reply:
Yeah I don't think there's a better way given we want to make the manim side as clean as possible (without unnecessary lines to fade to the color it already is)